### ISSUE TITLE
Intellij marks ZMongo usages as errors

### ISSUE THREAD

**Issue by brndt** [Kirill5k/mongo4cats#34]
I know Intellij is buggy sometimes when working with Scala 3 but can we somehow bypass the display of these errors? I suppose it ocurres because Intellij finds object and trait in the package simultaneously and cannot decide which one it is referring to :)

<img width="1044" alt="Screenshot 2023-08-22 at 21 15 34" src="https://github.com/Kirill5k/mongo4cats/assets/21007364/c0c61088-02eb-4dff-9248-13f69eebabba">
<img width="1154" alt="Screenshot 2023-08-22 at 21 12 43" src="https://github.com/Kirill5k/mongo4cats/assets/21007364/2dfdee46-37d0-4f11-88bc-c8dca0d5107d">

**Comment by Kirill5k** [comment]
The error that you have on the first screenshot is correct: `ZMongoClient` does not take any type arguments. it should just be:
```
database <- ZIO.serviceWithZIO[ZMongoClient](_.getDatabase(config.database))
```

The second error is quite weird though, Intellij shouldn't be complaining about it.

Out of curiosity, if you run code from [this example](https://github.com/Kirill5k/mongo4cats/blob/master/examples/src/main/scala/mongo4cats/examples/Zio.scala), will your IDE show errors there as well?

**Comment by brndt** [comment]
> The error that you have on the first screenshot is correct: ZMongoClient does not take any type arguments. it should just be

Yes I know, I just added two parameters in order to show that Intellij is behaving weird saying opposite things in the hint.
If you don't pass any parameters, it says you should add one; if you pass one, it says you shouldn't pass any; if you pass two, it says both things.

> Out of curiosity, if you run code from [this example](https://github.com/Kirill5k/mongo4cats/blob/master/examples/src/main/scala/mongo4cats/examples/Zio.scala), will your IDE show errors there as well?

Yes:
<img width="951" alt="Screenshot 2023-08-26 at 17 39 52" src="https://github.com/Kirill5k/mongo4cats/assets/21007364/14f0f83c-cdfb-4860-91f7-f4a18d6e9493">

**Comment by CodeCombiner** [comment]
Having the same issue on the latest Idea

**Comment by brndt** [comment]
https://youtrack.jetbrains.com/issue/SCL-21547
